Extension/src/Debugger/configurationProvider.ts

```diff
@@ -175,17 +130,48 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
                 config.debuggerEvent = DebuggerEvent.debugPanel;
             }
         }
-        if ((!folder || isIntelliSenseDisabled) && config.preLaunchTask) {
-            /* There are two cases where folder is undefined:
-                * when debugging is done on a single file where there is no folder open,
-                * or when the debug configuration is defined at the User level.
-                */
-            await this.resolvePreLaunchTask(undefined, config);
-            config.preLaunchTask = undefined;
-        } else {
-            await this.resolvePreLaunchTask(folder, config);
+
+        /** If the config is coming from "Run and Debug" debugPanel, there are three cases where the folder is undefined:
+         * 1. when debugging is done on a single file where there is no folder open,
+         * 2. when the debug configuration is defined at the User level (global).
+         * 3. when the debug configuration is defined at the workspace level.
+         * If the config is coming from "Run and Debug" playButton, there is one case where the folder is undefined:
+         * 1. when debugging is done on a single file where there is no folder open.
+         */
+
+        /** Do not resolve PreLaunchTask for these two cases, and let the VsCode resolve it:
+         * 1: The configs that comes from playButton and are already defined, these configs should already have their PreLaunchTask defined.
```

Comment:
Missing word after "and" in the sentence "The configs that comes from playButton and are already defined, these configs should already have their PreLaunchTask defined."